### DISCUSSION TITLE
How to support mongoDB projections along with pydantic response model.

### DISCUSSION THREAD

**Discussion by sree1026** [fastapi/fastapi#11319] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class User(BaseModel):
    name: Optional[str, None] 
    age: Optional[int, None]
    
@app.get('/user', response_model=User)
async def get_user(fields: Query(default="")):
    # using mongoDB projections to return only requested data
    # For ex: I'm requesting for name field using fields query
    return {
        "name": "ABC"
    }
```

### Description

I would like to know if there is way to prevent pydantic or fastAPI from adding missing fields with default values when serializing the response data but stick with validating only the fields that are present in the response data. 

To give a context, I am trying to implement field projections in GET API endpoint using mongoDB $projections. I already have a pydantic model which I is defined as response_model for that API endpoint. When we use field projections, the data returned from mongoDB would contain only the fields that were requested. This will throw validation error if I don't specify all the fields as Optional in the pydantic model. But if we specify that all the fields are Optional, then the response would contain default values for the fields that are missing which defeats the purpose of field projections.

I can create another API endpoint for this purpose alone and set response_model to None. This would perfectly work but the OpenAPI schema will not contain any response model for this API endpoint.

Please share if there is any recommended robust way to handle this. 

PS: The code I had shared is just a high level idea of what I want. Please let me know if working version is needed. Also, I am using older versions of FastAPI and pydantic but I will be updating them soon.

### Operating System

Windows

### FastAPI Version

0.95.1

### Pydantic Version

1.10.7

### Python Version

3.8.10

**Comment by YuriiMotov** [comment, marked as answer]
If I understand you question correctly, you just need to use `response_model_exclude_unset=True`:
```
from typing import Optional
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class User(BaseModel):
    name: Optional[str]  = None
    age: Optional[int] = None
    
@app.get('/user', response_model=User, response_model_exclude_unset=True)
async def get_user():
    return {
        "name": "ABC"
    }
```

**Reply by sree1026** [reply]
Yes, exactly thats what I was looking for. I have been searching it for few days, going over multiple stackoverflow questions but just a configuration change did the job. Thanks a lot.
